hyperspy/io_plugins/usid_hdf5.py

```diff
@@ -0,0 +1,447 @@
+                                         'Supply keyword argument "ignore_non_linear_dims=True" to ignore this '
+                                         'error'.format(dim_name))
+
+        step_size = step_size[0]
+        dim_dict[dim_name] = {'size': len(dim_vals),
+                              'name': dim_name,
+                              'units': units,
+                              'scale': step_size,
+                              'offset': dim_vals[0]}
+    return dim_dict
+
+
+def __assemble_dim_list(dim_dict, dim_names):
+    """
+    Assembles a list of dictionary objects (axes) in the same order as specified in dim_names
+
+    Parameters
+    ----------
+    dim_dict : dict
+        Dictionary of dictionaries that correspond to axes for HyperSpy Signal objects
+    dim_names : list
+        List of strings denoting the names of the dimension
+
+    Returns
+    -------
+    list
+        List of dictionaries that correspond to axes for HyperSpy Signal objects
+    """
+    dim_list = []
+    for dim_name in dim_names:
+        try:
+            dim_list.append(dim_dict[dim_name])
+        except KeyError:
+            pass
+    return dim_list
+
+
+def __split_descriptor(desc):
+    """
+    Splits a string such as "Quantity [units]" or "Quantity (units)" into the quantity and unit strings
+
+    Parameters
+    ----------
+    desc : str
+        Descriptor of a dimension or the main dataset itself
+
+    Returns
+    -------
+    quant : str
+        Name of the physical quantity
+    units : str
+        Units corresponding to the physical quantity
+    """
+    desc = desc.strip()
+    ind = desc.rfind('(')
+    if ind < 0:
+        ind = desc.rfind('[')
+        if ind < 0:
+            return desc, ''
+
+    quant = desc[:ind].strip()
+    units = desc[ind:]
+    units = units.replace('(', '')
+    units = units.replace(')', '')
+    units = units.replace('[', '')
+    units = units.replace(']', '')
+    return quant, units
+
+
+def __convert_to_signal_dict(ndim_form, quantity, units, dim_dict_list,
+                             h5_path, h5_dset_path, sig_type=''):
+    """
+    Packages required components that make up a Signal object
+
+    Parameters
+    ----------
+    ndim_form : numpy.ndarray
+        N-dimensional form of the main dataset
+    quantity : str
+        Physical quantity of the measurement
+    units : str
+        Corresponding units
+    dim_dict_list : list
+        List of dictionaries that instruct the axes corresponding to the main dataset
+    h5_path : str
+        Absolute path of the original USID HDF5 file
+    h5_dset_path : str
+        Absolute path of the USIDataset within the HDF5 file
+    sig_type : str
+        Type of measurement
+
+    Returns
+    -------
+
+    """
+
+    sig = {'data': ndim_form,
+           'axes': dim_dict_list,
+           'metadata': {
+               'Signal':{'quantity': quantity, 'units': units, 'signal_type': sig_type},
```

Comment:
`units` is not a metadata as defined in the metadata structure: http://hyperspy.org/hyperspy-doc/current/user_guide/metadata_structure.html

Reply:
OK. I moved all non-standard metadata to original_metadata